deployment/terraform/modules/osv/website.tf

```diff
@@ -32,4 +33,122 @@ resource "google_cloud_run_service_iam_binding" "website" {
   ]
 }
 
-# TODO: Set up Google Cloud Load Balancing + Network Endpoint Group (NEG)
\ No newline at end of file
+# SSL Certificates
+resource "google_certificate_manager_dns_authorization" "website" {
+  project     = var.project_id
+  name        = "website-dnsauth"
+  description = "The dns auth for the osv.dev website"
+  domain      = var.website_domain
+}
+
+resource "google_certificate_manager_certificate" "website" {
+  project     = var.project_id
+  name        = "website-cert"
+  description = "The osv.dev website cert"
+  managed {
+    domains = [var.website_domain, "*.${var.website_domain}"]
+    dns_authorizations = [
+      google_certificate_manager_dns_authorization.website.id
+    ]
+  }
+  lifecycle {
+    replace_triggered_by = [google_certificate_manager_dns_authorization.website.id]
+  }
+}
+
+resource "google_certificate_manager_certificate_map" "website" {
+  project     = var.project_id
+  name        = "website-certmap"
+  description = "osv.dev website certificate map"
+}
+
+resource "google_certificate_manager_certificate_map_entry" "website" {
+  project      = var.project_id
+  name         = "website-certmap-entry"
+  description  = "osv.dev website certificate map entry"
+  map          = google_certificate_manager_certificate_map.website.name
+  certificates = [google_certificate_manager_certificate.website.id]
+  hostname     = var.website_domain
+
+  lifecycle {
+    replace_triggered_by = [google_certificate_manager_certificate_map.website.id, google_certificate_manager_certificate.website.id]
+  }
+}
+
+# Load Balancer
+module "gclb" {
+  source  = "terraform-google-modules/lb-http/google//modules/serverless_negs"
+  version = "~> 10.0"
+
+  name    = "website"
+  project = var.project_id
+
+  enable_ipv6         = true
+  create_ipv6_address = true
+  ssl                 = true
+  certificate_map     = google_certificate_manager_certificate_map.website.id
+
+  load_balancing_scheme = "EXTERNAL_MANAGED"
+
+  backends = {
+    default = {
+      groups = [
+        {
+          # TODO(michaelkedar): Replace with Cloud Run when ready
```

Comment:
The picture I had in my head was we could have both back ends ~now, with the Cloud Run backend drained of any traffic. Is that possible to do?

Reply:
I think I've managed to work out how to do this with c650a4d